services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/quarkus/CommonConfigPropertiesProducer.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.deviceregistry.quarkus;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.eclipse.hono.client.amqp.config.ClientConfigProperties;
+import org.eclipse.hono.client.amqp.config.ClientOptions;
+import org.eclipse.hono.client.kafka.CommonKafkaClientOptions;
+import org.eclipse.hono.client.kafka.producer.KafkaProducerOptions;
+import org.eclipse.hono.client.kafka.producer.MessagingKafkaProducerConfigProperties;
+import org.eclipse.hono.client.notification.kafka.NotificationKafkaProducerConfigProperties;
+import org.eclipse.hono.deviceregistry.service.device.AutoProvisionerConfigOptions;
+import org.eclipse.hono.deviceregistry.service.device.AutoProvisionerConfigProperties;
+import org.eclipse.hono.service.auth.delegating.AuthenticationServerClientConfigProperties;
+import org.eclipse.hono.service.auth.delegating.AuthenticationServerClientOptions;
+
+import io.smallrye.config.ConfigMapping;
+
+/**
+ * A producer of registry service configuration properties commonly used in registry implementations.
+ *
+ */
+@ApplicationScoped
+public class CommonConfigPropertiesProducer {
+
+    @Produces
+    @Singleton
+    @Named("amqp-messaging-network")
+    ClientConfigProperties downstreamSenderProperties(
+            @ConfigMapping(prefix = "hono.messaging", namingStrategy = ConfigMapping.NamingStrategy.VERBATIM)
+            final ClientOptions downstreamSenderOptions) {
+        final var result = new ClientConfigProperties(downstreamSenderOptions);
+        result.setServerRoleIfUnknown("AMQP Messaging Network");
+        result.setNameIfNotSet("Hono Device Registry");
```

Comment:
Using a common name here (instead of "Hono JDBC Device Registry"), so that a common method can be used.
I think that is ok here.
(I've experimented a bit with a "componentName" bean, but I think it's not worth the overhead here.)